src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala

```diff
@@ -293,13 +294,30 @@ object AOIMonitorUtils extends Logging{
     ???
   }
 
-  def queryLastSequence(databaseURI: URI, procName: String): Option[Int] = {
+  def getLastSequence(interval: Interval): Option[Int] = {
+    queryBeginSequence(interval)
+  }
+
+  def getCurrentSequence(augmentedDiffSource: URI, interval: Interval)(
+      implicit spark: SparkSession): Option[Int] = {
+    spark.sparkContext
+      .parallelize(Seq(augmentedDiffSource))
+      .map { uri =>
+        queryBeginSequence(interval)
+      }
+      .collect
+      .apply(0)
+  }
+
+  def warnMessage: (=> String) => Unit = logWarning
+
+  def queryBeginSequence(interval: Interval): Option[Int] = {
```

Comment:
All of the significant changes for this PR are from this line down to the bottom of the file.